Boolean_set_operations_2/doc/Boolean_set_operations_2/CGAL/Boolean_set_operations_2.h

```diff
@@ -1,833 +1,2166 @@
+                  const General_polygon_2<ArrTraits>& pgn2);
+
+/*! determines whether two general polygons intersect in their interior.
+ * \param pgn1 the 1st input polygon.
+ * \param pgn2 the 2nd input polygon.
+ * \return `true` if `pgn1` and `pgn2` intersect in their interiro and `false`
+ *         otherwise.
+ * \pre `%ArrTraits` must be a model of the concept
+ *      `ArrangementDirectionalXMonotoneTraits_2`.
  */
-template <class Traits>
-bool do_intersect(const General_polygon_with_holes_2<General_polygon_2<Traits> > & p1,
-                  const General_polygon_2<Traits> & p2);
-
-
-/*!
-  returns `true` if the general polygons `p1` and `p2` intersect in their interior.
+template <typename ArrTraits>
+bool
+do_intersect(const General_polygon_2<ArrTraits>& pgn1,
+             const General_polygon_with_holes_2<General_polygon_2<ArrTraits> >& pgn2);
+
+/*! determines whether two general polygons intersect in their interior.
+ * \param pgn1 the 1st input polygon.
+ * \param pgn2 the 2nd input polygon.
+ * \return `true` if `pgn1` and `pgn2` intersect in their interiro and `false`
+ *         otherwise.
+ * \pre `%ArrTraits` must be a model of the concept
+ *      `ArrangementDirectionalXMonotoneTraits_2`.
  */
-template <class Polygon>
-bool do_intersect(const General_polygon_with_holes_2<Polygon> & p1,
-                  const General_polygon_with_holes_2<Polygon> & p2);
-
-  /*!
-    returns `true`, if the set of general polygons (or general
-    polygons with holes) in the given range intersect in their interior,
-    and `false` otherwise. (The value type of the input iterator is
-    used to distinguish between the two).
-  */
-template <class InputIterator>
+template <typename ArrTraits>
+bool do_intersect(const General_polygon_with_holes_2<General_polygon_2<ArrTraits> >& pgn1,
+                  const General_polygon_2<ArrTraits>& pgn2);
+
+/*! determines whether two general polygons with holes intersect in their
+ * interior.
+ * \param pgn1 the 1st input polygon.
+ * \param pgn2 the 2nd input polygon.
+ * \return `true` if `pgn1` and `pgn2` intersect in their interiro and `false`
+ *         otherwise.
+ */
+template <typename Polygon>
+bool do_intersect(const General_polygon_with_holes_2<Polygon>& pgn1,
+                  const General_polygon_with_holes_2<Polygon>& pgn2);
+
+/*! Given a range of polygons or a range of polygons with holes (resp. a range
+ * of general polygons or a range of general polygons with holes) determines
+ * whether the open polygons (resp. general polygons) in the range have a common
+ * point.
+ * \param begin the first iterator of the input range. It's value type is
+ *        either Polygon_2 (resp. General_polygon_2) or Polygon_with_holes
+ *        (resp. General_polygon_with_holes_2).
+ * \param end the past-the-end iterator of the input range. It's value type is
+ *        either Polygon_2 (resp. General_polygon_2) or Polygon_with_holes
+ *        (resp. General_polygon_with_holes_2).
```

Comment:
```suggestion
 *        (resp. `General_polygon_with_holes_2`).
```

Reply:
There are many similar missing backticks, I won't mark them all in the review...